workbook/views/views.py

```diff
@@ -123,3 +124,110 @@ def get(self, request):
         result = self.worker_service.search(params)
 
         return Response(result, status=status.HTTP_200_OK)
+
+
+class CustomerReservationsView(APIView):
+    def __init__(self):
+        self.reservation_service = ReservationService()
+        self.time_service = TimeService()
+
+    def get(self, request, customer_id):
+        reservations = self.reservation_service.get_customer_reservations(customer_id)
+
+        for reservation in reservations:
+            reservation.start_date_time = self.time_service.unix_to_datetime(reservation.start_date_time)
+
+        reservations_serializer = CustomerReservationResponseSerializer(reservations, many=True)
+
+        return Response(reservations_serializer.data, status=status.HTTP_200_OK)
+
+    def post(self, request, customer_id):
+        data = request.data
+        serializer = CreateReservationSerializer(data=data)
+        serializer.is_valid_raise()
+        reservation_ids = self.reservation_service.create(customer_id, serializer)
+
+        if reservation_ids:
+            return Response(reservation_ids, status=status.HTTP_200_OK)
+        else:
+            raise BadRequest("Failed to create reservation.")
+
+
+class CustomerReservationView(APIView):
+    def __init__(self):
+        self.reservation_service = ReservationService()
+        self.time_service = TimeService()
+
+    def get(self, request, reservation_id, customer_id):
+        reservation = self.reservation_service.get_customer_reservation(reservation_id, customer_id)
+
+        reservation.start_date_time = self.time_service.unix_to_datetime(reservation.start_date_time)
```

Comment:
this is serializer duty, this will trigger an update statement to the db in some cases, and it is also not correct imagine doing it then accessing the data elsewhere it would be wrong.